parsl/monitoring/db_manager.py

```diff
@@ -310,40 +343,69 @@ def start(self, priority_queue, node_queue, resource_queue) -> None:
                                 reprocessable_first_resource_messages.append(
-                                    deferred_resource_messages.pop(msg['task_id']))
+                                    deferred_resource_messages.pop(task_try_id))
+                    else:
+                        raise RuntimeError("Unexpected message type {} received on priority queue".format(msg_type))
 
-                logger.debug(
-                    "Updating and inserting TASK_INFO to all tables")
+                logger.debug("Updating and inserting TASK_INFO to all tables")
+                logger.debug("Updating {} TASK_INFO into workflow table".format(len(task_info_update_messages)))
+                self._update(table=WORKFLOW,
+                             columns=['run_id', 'tasks_failed_count',
+                                      'tasks_completed_count'],
+                             messages=task_info_all_messages)
 
                 if task_info_insert_messages:
                     self._insert(table=TASK, messages=task_info_insert_messages)
                     logger.debug(
                         "There are {} inserted task records".format(len(inserted_tasks)))
+
                 if task_info_update_messages:
-                    self._update(table=WORKFLOW,
-                                 columns=['run_id', 'tasks_failed_count',
-                                          'tasks_completed_count'],
-                                 messages=task_info_update_messages)
+                    logger.debug("Updating {} TASK_INFO into task table".format(len(task_info_update_messages)))
                     self._update(table=TASK,
-                                 columns=['task_time_submitted',
+                                 columns=['task_time_invoked',
                                           'task_time_returned',
                                           'run_id', 'task_id',
-                                          'task_fail_count',
-                                          'task_fail_history'],
+                                          'task_fail_count'],
                                  messages=task_info_update_messages)
+                logger.debug("Inserting {} task_info_all_messages into status table".format(len(task_info_all_messages)))
+
                 self._insert(table=STATUS, messages=task_info_all_messages)
 
+                if try_insert_messages:
+                    logger.debug("Inserting {} TASK_INFO to try table".format(len(try_insert_messages)))
+                    self._insert(table=TRY, messages=try_insert_messages)
+                    logger.debug(
+                        "There are {} inserted task records".format(len(inserted_tasks)))
+
+                if try_update_messages:
+                    logger.debug("Updating {} TASK_INFO into try table".format(len(try_update_messages)))
+                    self._update(table=TRY,
+                                 columns=['task_try_time_returned',
```

Comment:
There are two `task_try_time_returned` in `columns`

Reply:
should be fixed in 11eb6c968